sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -93,6 +93,20 @@ TEST(Context, IsCanceled)
   EXPECT_TRUE(c2.IsCanceled());
 }
 
+TEST(Context, IsCanceledAfter)
+{
+  auto duration = std::chrono::milliseconds(150);
+  auto deadline = std::chrono::system_clock::now() + duration;
+
+  Context context;
+  auto c2 = context.WithDeadline(deadline);
+  EXPECT_NO_THROW(c2.ThrowIfWillCancelledAfter(std::chrono::milliseconds(149)));
+  EXPECT_THROW(c2.ThrowIfWillCancelledAfter(std::chrono::milliseconds(150)), OperationCanceledException);
```

Comment:
You can have a private or internal function in order to unit test (if you desire).